### ISSUE TITLE
[Feature] Support basic auth in iOS app

### ISSUE THREAD

**Issue by com6056** [TheSpaghettiDetective/obico-server#667]
**Is your feature request related to a problem? Please describe.**
Obico iOS app doesn't support Obico self-hosted servers with basic authentication in front of them.

**Describe the solution you'd like**
I'd love a way to input basic authentication credentials in the iOS app when logging into a self-hosted Obico server.

**Describe alternatives you've considered**
I've considered disabling my authentication layer that I have in front of Obico, but I'd be much more comfortable being able to keep it.

**Additional context**
Attempting to use basic authentication credentials in the URL (`https://username:password@example.com`) breaks the app and all I get is a blank white screen when opening it afterwards until I delete it and re-install it.

**Comment by kennethjiang** [comment]
Currently, the app would  ask you to login again only when you haven't logged in for 3 months. Isn't this a good enough alternative?

**Comment by com6056** [comment]
I'd like to avoid exposing Obico to the open internet with only it's built-in authentication, I'd be much more comfortable with keeping my own authentication layer in front of it. If the app could just allow putting in basic auth creds, storing them, and passing those along in the requests, that should solve it.

**Comment by kennethjiang** [comment]
Do you mean the basic auth *in addition to*, not "instead of" the built-in auth?

**Comment by com6056** [comment]
Yep, just want the ability to pass basic auth creds in each request sent by the iOS client 🙌 No need to remove anything
